File: backend/src/Business/Entities/LimiteOrcamento.cs
using System.ComponentModel.DataAnnotations;

namespace Business.Entities
{
    public class LimiteOrcamento : Entity
    {
        public Guid? CategoriaId { get; set; }
        
        public string UsuarioId { get; set; } = null!;

        public DateOnly Periodo { get; set; } 

        public decimal Limite { get; set; }

        public TipoLimite TipoLimite { get; set; }

        public decimal PorcentagemAviso { get; set; }

        // Props navigation ef
        public Categoria? Categoria { get; set; }

        public Usuario Usuario { get; set; } = null!;

    }

    public enum TipoLimite
    {
        Geral = 1,
        Categoria
    }
}

File: front-end/src/app/models/limiteOrcamento.ts
export interface LimiteOrcamento {
  id: string
  categoriaId?: string
  categoriaNome?: string
  periodo: Date
  limite: number
  tipoLimite: TipoLimite
  porcentagemAviso: number
  limiteUtilizado?: number
  percentualLimiteUtilizado?: string
}

export enum TipoLimite
{
  geral = 1,
  categoria
}